app/code/core/Mage/Core/Controller/Varien/Front.php

```diff
@@ -182,6 +182,12 @@ public function dispatch()
         Mage::dispatchEvent('controller_front_send_response_before', array('front'=>$this));
         Varien_Profiler::start('mage::app::dispatch::send_response');
         $this->getResponse()->sendResponse();
+        //This function flushes all response data to the client and finishes the request.
+        // This allows for time consuming tasks to be performed without leaving the connection
+        // to the client open.
+        if(php_sapi_name() == 'fpm-fcgi' && true === function_exists('fastcgi_finish_request')){
+            fastcgi_finish_request();
+        }
         Varien_Profiler::stop('mage::app::dispatch::send_response');
         Mage::dispatchEvent('controller_front_send_response_after', array('front'=>$this));
```

Comment:
Maybe this should go after line 186 in case for some reason someone wants to inject some additional content like an html comment with cache debug info or similar?